### ISSUE TITLE
WP Notification Center + certain types of admin notices breaks our editor, Matt becomes sad.

### ISSUE THREAD

**Issue by mathetos** [CalderaWP/Caldera-Forms#934]
Admin scripts seem to be enqueued in an order which breaks occassionally. This results in all the fields of the form in ADMIN going white and everything being uneditable.

We've manually rearranged scripts a few times to get it working, but it breaks again on every update.

![image](https://cloud.githubusercontent.com/assets/930724/20288998/11c0fbfe-aa8c-11e6-89e5-ebd92659b377.png)

Here's the Form edit screen:

![image](https://cloud.githubusercontent.com/assets/930724/20289044/51dd4698-aa8c-11e6-95da-be1a676b1b9b.png)

**Comment by Shelob9** [comment]
I don't know why order your referring to. But the error your showing normally means that the intial PHP rendered editor is not valid for Handlebars.js to parse.

This typically happens when an error is printed in the middle of the HTML source. If you can share any potential PHP errors you can catch with WP_DEBUG_LOG, that would be helpful.

**Comment by mathetos** [comment]
I'm digging into it more today. unfortunately it only happens on the live site, not locally. I'll keep you posted though.

**Comment by mathetos** [comment]
So in the end, this was a plugin conflict, but a very unique constellation of conflicts.

1) EDD with a Paid Extension that is expired so that their licensing alert appears
2) The free Notifications Center plugin
3) Plus Yoast SEO

That combo is what is breaking the DOM. 

Here's the combo that was triggering the breakage:

1) EDD + Notification Center == Fine.
2) EDD + NC + Yoast = Broke
3) EDD + Yoast = Fine

So it seems it's likely a combo of the alerts going into NC from either EDD licensing or Yoast or both.

**Comment by mathetos** [comment]
Reported it with Notification Center here:
https://wordpress.org/support/topic/breaks-dom-with-yoast-andor-edd-licensing-alerts/

Maybe @pippinsplugins  or @jdevalk have some insight into that.

**Comment by pippinsplugins** [comment]
Well that's weird.

For the EDD licensing alerts, are you referring to the one that shows when you have an invalid or expired license?

**Comment by jdevalk** [comment]
@tacoverdo might be able to help debug :)

**Comment by mathetos** [comment]
@pippinsplugins Yes, those ones. 

I'll say though that even though what I found above was very clearly what affected our live site, I have not been able to replicate it in other environments yet. Tried with the exact same CF Addons, EDD Addons, Yoast version... but it's clear that the DOM was broken right around where the EDD Notices were inserted into the page.

**Comment by tacoverdo** [comment]
Do you have a dump of the source of the broken page?
If so, what changes do you see when you disable Yoast SEO or the Notification Center? Which changes could be relevant in this case?

I've been trying to replicate this issue locally, but have been unable to do so.

**Comment by mathetos** [comment]
I'll PM you the source dump on FB, would rather not have it here since it's a public repo.

1. If I disable Yoast and have the Notification Center still active, the problem is resolved
2. If I disable Notification Center but keep Yoast Active the problem is resolved -- which is the current state of our site since Yoast is necessary and NC is just nice to have

I also just discovered that it's only breaking SOME forms. Not all, which brings it back to potentially a CF issue. It seems it's breaking forms that have columns. But again... I tried this exact conglomeration of plugins and the exact form in a different environment and couldn't reproduce there either.

**Comment by Shelob9** [comment]
I'm very tempted to add something like $('.notices'). remove(): to the JS
on our editor page...

On Nov 22, 2016 4:35 PM, "Matt Cromwell" <notifications@github.com> wrote:

I'll PM you the source dump on FB, would rather not have it here since it's
a public repo.

   1. If I disable Yoast and have the Notification Center still active, the
   problem is resolved
   2. If I disable Notification Center but keep Yoast Active the problem is
   resolved -- which is the current state of our site since Yoast is necessary
   and NC is just nice to have

I also just discovered that it's only breaking SOME forms. Not all, which
brings it back to potentially a CF issue. It seems it's breaking forms that
have columns. But again... I tried this exact conglomeration of plugins and
the exact form in a different environment and couldn't reproduce there
either.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/CalderaWP/Caldera-Forms/issues/934#issuecomment-262372699>,
or mute the thread
